src/shell.rs

```diff
@@ -1484,6 +1498,13 @@ fn init_nvim_async(
     rows: i32,
 ) {
     let nvim_result = if let Some(addr) = options.server {
+        #[cfg(unix)]
+        if let Ok(addr) = addr.parse() {
+            nvim::start_tcp_client(nvim_handler, addr, *options.timeout)
+        } else {
+            nvim::start_unix_socket_client(nvim_handler, addr, *options.timeout)
+        }
```

Comment:
Either way, IMHO we should move this argument parsing into `src/main.rs` so we can at least try opening the socket/unix socket and error-out before going through the whole startup process.